File: ABC038_C.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int N, a;
    vector<int> A;

    cin >> N;
    for (int i = 0; i < N; i++){
        cin >> a;
        A.push_back(a);
    }

    int left=0, right=0;
    long long ans=0;

    for (left=0; left < N; left++) {
        if (left == right) {
            right++;
        }
        while(right < N && A.at(right - 1) < A.at(right)) {
            right++;
        }
        ans += right - left;
    }

    cout << ans << endl;

}

File: ABC038_C.py
import sys

N = int(input())
A = list(map(int, sys.stdin.readline().strip().split()))

ans = 0
left = 0
right = 1

while right <= N:
    if right == N or A[right - 1] >= A[right]:
        n = right - left + 1
        ans += (n * (n - 1)) // 2

        left = right
    right += 1


print(ans)
